gp-templates/helper-functions.php

```diff
@@ -344,6 +344,25 @@ function textareas( $entry, $permissions, $index = 0 ) {
 		?>
 		<blockquote class="translation"><?php echo prepare_original( esc_translation( gp_array_get( $entry->translations, $index ) ) ); ?></blockquote>
 		<textarea class="foreign-text" name="translation[<?php echo esc_attr( $entry->original_id ); ?>][]" id="translation_<?php echo esc_attr( $entry->original_id ); ?>_<?php echo esc_attr( $index ); ?>" <?php echo disabled( ! $can_edit ); ?>><?php echo gp_prepare_translation_textarea( esc_translation( gp_array_get( $entry->translations, $index ) ) ); ?></textarea>
+		<div id="counts_<?php echo esc_attr( $entry->original_id ); ?>_<?php echo esc_attr( $index ); ?>" class="counts">
+			<?php
+			// Characters count.
+			$characters_count = mb_strlen( gp_array_get( $entry->translations, $index ) );
+			$characters = sprintf(
+				/* translators: %d: Number of characters. */
+				_n( '%d Character', '%d Characters', $characters_count, 'glotpress' ),
+				esc_html( $characters_count )
+			);
+			// Words count.
+			$words_count = count( preg_split( '/[\s]+/i', gp_array_get( $entry->translations, $index ), 0, PREG_SPLIT_NO_EMPTY | PREG_SPLIT_DELIM_CAPTURE ) );
```

Comment:
Do we have to replace newline characters first to avoid entries with only a single space? Looking at https://github.com/WordPress/wordpress-develop/blob/0d7487bad97e185c246e48be5a24fab5ee7bb985/src/wp-includes/formatting.php#L3950-L3951.

Reply:
This also doesn't handle languages where word count is based on single characters.

Reply:
> Do we have to replace newline characters first to avoid entries with only a single space?

Doesn't `\s` already match `\n`, `\r` and `\t`?

> This also doesn't handle languages where word count is based on single characters.

For that GlotPress needs to rely on the site language (core dev .po), should we go for it?

Reply:
@ocean90 what about completely remove the PHP fill of this data, and add it only on focus/update the current editor textarea with the `Wordcount` JS package?

Reply:
Removed the PHP logic and relying completely on the the Wordcount package to update the data on textarea focus or any input change. This reduces the PHP processing for all the strings and only processes the count on opening the row editor.